src/com/company/Game.java

```diff
@@ -0,0 +1,258 @@
+package com.company;
+
+import java.util.ArrayList;
+
+public class Game {
+
+    ArrayList<Team> teams= new ArrayList<>();
+    int overs = 6;
+    public void setupTeams(){
+        Team team1 = new Team();
+        Team team2 = new Team();
+        team1.setId(1001);
+        team2.setId(1002);
+        team1.setName("CSK");
+        team2.setName("MI");
+        team1.setPlayers(11) ;
+        team2.setPlayers(11) ;
+        team1.setNotOut(11);
+        team2.setNotOut(11);
+        System.out.println("Team 1 name: " + team1.getName());
+        System.out.println("Team 2 name: " + team2.getName());
+        team1.setupPlayers(101);
+        team2.setupPlayers(201);
+        teams.add(team1);
+        teams.add(team2);
+
+    }
+
+
+
+    public void match()
+    {
+        //int team1_w = team1.getPlayers(), team2_w = team2.getPlayers();
+        int totalBalls = overs*6;
+        int tossVar = toss();
+        if(tossVar==1)
+        {
+            Team t = teams.get(0);
+            teams.remove(0);
+            teams.add(t);
+        }
+        System.out.println("Toss is WON by: "+teams.get(0).getName()+" and they choose to bat");
+        System.out.println("");
+        //int strike=0,nonStrike=0,playerNext=3;
+        //team 1 batting starts
+        onCrease strike = new onCrease(0,0,0,1);
```

Comment:
onCrease class is unnecessary.  it is just acting act a temporary storage of maintaining runs scored and balls faced. at the end you are setting those variables in player class. 
why then not maintain two instances, onStrike batsman and nonStrike batsman and update them.

Reply:
Removed the onCrease class and utilised the player class only. As suggested, used onStrike, nonStrkie for batting and bowler for bowling.